amqtt/mqtt/protocol/client_handler.py

```diff
@@ -106,17 +106,18 @@ async def mqtt_subscribe(self, topics, packet_id):
         # Wait for SUBACK is received
         waiter = futures.Future()
         self._subscriptions_waiter[subscribe.variable_header.packet_id] = waiter
-        return_codes = await waiter
-
-        del self._subscriptions_waiter[subscribe.variable_header.packet_id]
+        try:
+            return_codes = await waiter
+        finally:
+            del self._subscriptions_waiter[subscribe.variable_header.packet_id]
         return return_codes
 
     async def handle_suback(self, suback: SubackPacket):
         packet_id = suback.variable_header.packet_id
-        try:
-            waiter = self._subscriptions_waiter.get(packet_id)
+        waiter = self._subscriptions_waiter.get(packet_id)
```

Comment:
The previous code was looking for a KeyError when there is no possible way to raise it. I expect this was leftover from
self._subscriptions_waiter[packet_id] and then wasn't updated when it was changed to self._subscriptions_waiter.get(packet_id)